### PR TITLE
Add pydantic.v1 namespace to Pydantic v1

### PR DESCRIPTION
**Pull request by exs-dmiketa**
## Change Summary

Following jenshnielsen's brilliant idea in https://github.com/pydantic/pydantic/issues/6022, I added a `pydantic.v1` alias to the v1 version of Pydantic. This will enable cross-version Pydantic support and easier migration of expansive v1 codebases by switching all `pydantic` imports to `pydantic.v1` and removing the upper bound.

Fix https://github.com/pydantic/pydantic/issues/6022

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [ ] Tests pass on CI
* [ ] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @hramezani

skip change file check

### CODE REVIEW

**Comment by exs-dmiketa** [comment]
This is my first Pydantic PR so sincere apologies if I'm doing this wrong!

I added a very simple test to capture the main functionality.

Crucially, the following will _not_ work: `from pydantic.v1.dataclasses import dataclass`. But `from pydantic.v1 import dataclasses` will. We could try to fool the submodule detection in at least two ways:
1. Copy entire files into a `v1` subfolder. (Not massively different from the "v1 in v2" setup.)
2. Put dummy files into a `v1` subfolder, e.g. `v1/dataclasses.py`, and only reimport the top-level corresponding file (`from ..dataclasses import *`)

Given that this would be merely a transitional solution to help users move from v1 to v2, I'm not sure the issue is critical, but happy to defer to maintainers. Please review. :)

**Review by sydney-runkle — APPROVED** [review]
@exs-dmiketa,

Great, thanks! We'll release this change with an upcoming V1 patch (probably next week).

File: tests/test_v1.py
Comment (sydney-runkle) [inline tests/test_v1.py:2]:
```suggestion
    from pydantic.v1 import BaseModel, dataclasses  # type: ignore
```

File: tests/test_v1.py
Comment (sydney-runkle) [inline tests/test_v1.py:2]:
```suggestion
    from pydantic.v1 import BaseModel, dataclasses  # noqa: F401
```

**Comment by exs-dwoodward** [comment]
Hi all, this change is useful for things being imported directly like:

```python
from pydantic import ...
``` 

but for imports like:
```python
from pydantic.fields import ModelField
from pydantic.generics import GenericModel
```
or:
```python
from pydantic import fields
from pydantic import generics
```

it doesn't translate as well. @sydney-runkle would a PR that adds the ability to do these kinds of imports from the `v1.py` module be useful? IIUC that would make v1 --> v2 migration (with the same existing v1 types / models of course) simply turn into a find + replace of `from pydantic` --> `from pydantic.v1` in all cases. I have a branch locally with this and am happy to contribute a PR if you think it would be useful!

**Pull request merged**
